### PR TITLE
[SQL Definition Collation][Oracle]Collate the SQL definition of DROP TABLE

### LINKED ISSUE #9059: [SQL Definition Collation][Oracle]Collate the SQL definition of DROP TABLE
Target
- Collate SQL definition of DROP TABLE to make SQL definition of Apache ShardingSphere the same with Oracle's.
- Add a test case

HOW
1 Compare DROP TABLE definition in Oficial SQL Doc and ShardingSphere SQL Doc
2 If there is any difference in ShardingSphere SQL Doc, please correct them referring to Official SQL Doc (g4 file of ANTLR)
3 Run mvn install shardingsphere-sql-parser-oracle
4 Check whether there are any exceptions. If exists, please fix them. (Please pay attention to xxxVisitor.class)
5 Add new corresponding SQL case in shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/resources/sql and expected parsed result in shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/resources/case
6 Run SQLParserParameterizedTest (shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/java/org/apache/shardingsphere/test/sql/parser/parameterized/engine/SQLParserParameterizedTest.java).

Tips
- Since this issue is a child one of #6480, I guess the PRs relevant to #6480 will give you some examples or hints.

### PR DESCRIPTION
**Pull request by ThanoshanMV**
Changes proposed in this pull request:
- added SQL definition of `DROP TABLE`
- added a test case

### CODE REVIEW

**Comment by ThanoshanMV** [comment]
Hi @tristaZero, I've added the SQL definition of `DROP TABLE`. Can you please check it? 

I read [this issue](https://github.com/apache/shardingsphere/issues/4889) to understand the `start-index` and `stop-index`.

**Comment by tristaZero** [comment]
@ThanoshanMV Thanks for your quick PR, I guess a test will be added later, for now let leave @wgy8283335 some time to give it a look. 

Until now, some simple issues are picked up for you to have a basic learning about our [COMDEV-385](https://issues.apache.org/jira/browse/COMDEV-385) , which is much harder. Do you want to have some other little tasks or begin to try COMDEV-385? 

BTW, the timeline in (GSOC2021) [https://community.apache.org/gsoc.html#prospective-asf-mentors-read-this] shows us,

> We are now collecting ideas for GSoC 2021
> 2021-03-30: Student applications start

**Review by tristaZero — APPROVED** [review]
LGTM.

**Review by tristaZero — COMMENTED** [review]
Hi I gave a look at [DROP TABLE](https://docs.oracle.com/en/database/oracle/oracle-database/19/sqlrf/DROP-TABLE.html#GUID-39D89EDC-155D-4A24-837E-D45DDA757B45)
How about `schema.` in this definition?

**Review by tristaZero — CHANGES_REQUESTED** [review]
Hi I gave a look at DROP TABLE
How about `schema.` in this definition?

**Comment by ThanoshanMV** [comment]
Hi @tristaZero, I'll make the changes and let you know

**Comment by ThanoshanMV** [comment]
Hi @tristaZero, I've modified the definition. Can you please check it?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/resources/sql/supported/ddl/drop.xml [hunk 1]
```diff
@@ -23,6 +23,7 @@
     <sql-case id="drop_table_restrict" value="DROP TABLE t_log RESTRICT" db-types="MySQL,PostgreSQL" />
     <sql-case id="drop_table_cascade" value="DROP TABLE t_log CASCADE" db-types="MySQL,PostgreSQL" />
     <sql-case id="drop_table_cascade_constraints_and_purge" value="DROP TABLE t_log CASCADE CONSTRAINTS PURGE" db-types="Oracle" />
+    <sql-case id="drop_table_purge" value="DROP TABLE t_log PURGE" db-types="Oracle" />
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/resources/sql/supported/ddl/drop.xml:26]: I suppose `DROP TABLE from db PURGE` is a more valuable case. Could you add this one or replace that simple one?

**Review by tristaZero — CHANGES_REQUESTED** [review]

File: shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/resources/sql/supported/ddl/drop.xml (hunk 1 above)
Comment (ThanoshanMV) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/resources/sql/supported/ddl/drop.xml:26]: Okay, I'll do it @tristaZero

Comment (ThanoshanMV) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/resources/sql/supported/ddl/drop.xml:26]: `DROP TABLE db.name PURGE`: is it the query we need to specify there?

**Comment by ThanoshanMV** [comment]
> @ThanoshanMV Thanks for your quick PR, I guess a test will be added later, for now let leave @wgy8283335 some time to give it a look.
> 
> Until now, some simple issues are picked up for you to have a basic learning about our [COMDEV-385](https://issues.apache.org/jira/browse/COMDEV-385) , which is much harder. Do you want to have some other little tasks or begin to try COMDEV-385?
> 
> BTW, the timeline in (GSOC2021)[https://community.apache.org/gsoc.html#prospective-asf-mentors-read-this] shows us,
> 
> > We are now collecting ideas for GSoC 2021
> > 2021-03-30: Student applications start

Hi @tristaZero, since [COMDEV-385](https://issues.apache.org/jira/browse/COMDEV-385) is hard and student applications start from at the end of March, I would like to try it as it'll help me to further explore the task, finish small subtasks and enough time to understand the task before submitting the application. 

I would like to start with small subtasks of the main task and finish them.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 [hunk 2]
```diff
@@ -37,7 +37,7 @@ alterIndex
     ;
 
 dropTable
-    : DROP TABLE tableName
+    : DROP TABLE (schemaName DOT_)? tableName (CASCADE CONSTRAINTS)? (PURGE)?
```
Comment (wgy8283335) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:40]: I am wondering whether need add "(schemaName DOT_)?",  as tableName definition is below:
```
tableName
    : (owner DOT_)? name
```
@tristaZero I think it's better to remove  "(schemaName DOT_)?", are you agree with me?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/resources/sql/supported/ddl/drop.xml (hunk 1 above)
Comment (tristaZero) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-test/src/main/resources/sql/supported/ddl/drop.xml:26]: Hi sorry for my mistake. You're right.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 (hunk 2 above)
Comment (tristaZero) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:40]: Hi, If that' is the the definition of `tableName`, I agree to remove `(schemaName DOT_)?` from this `dropTable` definition.
@ThanoshanMV Could you give it a look?

**Comment by tristaZero** [comment]
> Hi @tristaZero, since COMDEV-385 is hard and student applications start from at the end of March, I would like to try it as it'll help me to further explore the task, finish small subtasks and enough time to understand the task before submitting the application.
I would like to start with small subtasks of the main task and finish them.

I prefer your choice. Preparation work helps us move it forward smoothly. The target definitions of `COMDEV-385` are `SELECT/UPDATE/DELETE/INSERT`. From my perspective, `SELECT` is the most difficult one. My suggestion is first to try `INSERT` or `DELETE`. 

Any questions, please be free to ping me or @wgy8283335  at your PR.

Another concern is that I am not sure whether `GSOC2021` allows pre-raise PRs. If not, we need to figure out how to do our preparation work.

**Comment by ThanoshanMV** [comment]
> > Hi @tristaZero, since COMDEV-385 is hard and student applications start from at the end of March, I would like to try it as it'll help me to further explore the task, finish small subtasks and enough time to understand the task before submitting the application.
> > I would like to start with small subtasks of the main task and finish them.
> 
> I prefer your choice. Preparation work helps us move it forward smoothly. The target definitions of `COMDEV-385` are `SELECT/UPDATE/DELETE/INSERT`. From my perspective, `SELECT` is the most difficult one. My suggestion is first to try `INSERT` or `DELETE`.
> 
> Any questions, please be free to ping me or @wgy8283335 at your PR.
> 
> Another concern is that I am not sure whether `GSOC2021` allows pre-raise PRs. If not, we need to figure out how to do our preparation work.

Thank you for guiding me @tristaZero. If `GSOC2021` allows us to pre-raise PRs on the main task then I would like to try on the `INSERT` statement, if I finished it successfully then I'll try other little tasks such as #9018 until I submit the application and eligible to work on the main task. 

If `GSOC2021` doesn't allow us to pre-raise PRs then for the preparation, I would like to try other little tasks which will help me to try our COMDEV-385 once I'm eligible to do. 

I'll get help from you and @wgy8283335.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4 (hunk 2 above)
Comment (ThanoshanMV) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DDLStatement.g4:40]: Okay, I'll revert the previous commit.

**Comment by tristaZero** [comment]
@ThanoshanMV 

So the key point here is to make sure the pre-PR issue. I glance through [GSOC2021](https://opensource.googleblog.com/2020/10/google-summer-of-code-2021-is-bringing.html) and did not find the exact answer? Could you give it a check? 
Maybe the last way for us is to email the organizer.

**Review by tristaZero — APPROVED** [review]

**Comment by tristaZero** [comment]
Nice job. @ThanoshanMV

**Comment by ThanoshanMV** [comment]
> @ThanoshanMV
> 
> So the key point here is to make sure the pre-PR issue. I glance through [GSOC2021](https://opensource.googleblog.com/2020/10/google-summer-of-code-2021-is-bringing.html) and did not find the exact answer? Could you give it a check?
> Maybe the last way for us is to email the organizer.

Okay Trista, I'll check it and let you know.

**Comment by tristaZero** [comment]
@ThanoshanMV 
If there is any hints, you can comment on our COMDEV-385.
Thanks.

**Comment by ThanoshanMV** [comment]
Hi @tristaZero, I've searched through the documents and I haven't found anything about pre-PRs. Seems like we can not pre-PR. I'm also not sure about this. 

For the preparation work, is there any other SQL definition other than `SELECT/UPDATE/DELETE/INSERT` which requires the same work and steps I need to follow as I would need to follow for `SELECT/UPDATE/DELETE/INSERT`? If not, I can do other preparation tasks which you assign me to do. 

I'm okay with any of your tasks and suggestions.

**Pull request merged**